internal/proto/volumegroup.proto

```diff
@@ -0,0 +1,111 @@
+  rpc ModifyVolumeGroupMembership(ModifyVolumeGroupMembershipRequest)
+      returns (ModifyVolumeGroupMembershipResponse) {}
+  // DeleteVolumeGroup RPC call to delete a volume group.
+  rpc DeleteVolumeGroup(DeleteVolumeGroupRequest)
+      returns (DeleteVolumeGroupResponse) {}
+  // CreateVolumeGroup RPC call to get a volume group.
+  rpc ControllerGetVolumeGroup(ControllerGetVolumeGroupRequest)
+      returns (ControllerGetVolumeGroupResponse) {}
+}
+// CreateVolumeGroupRequest holds the required information to
+// create a volume group
+message CreateVolumeGroupRequest {
+  // suggested name for volume group (required for idempotency)
+  // This field is REQUIRED.
+  string name = 1;
+
+  // params passed to the plugin to create the volume group.
+  // This field is OPTIONAL.
+  map<string, string> parameters = 2;
+
+  // Secrets required by the driver to complete the
+  // create request.
+  string secret_name = 3;
+  string secret_namespace = 4;
+
+  // Specify volume_ids that will be added to the volume group.
+  // This field is OPTIONAL.
+  repeated string volume_ids = 5;
+}
+// CreateVolumeGroupResponse holds the information to send when
+// volumeGroup is successfully created.
+message CreateVolumeGroupResponse {
+  // Contains the identifier of the newly created volume group.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+}
+// DeleteVolumeGroupRequest holds the required information to
+// delete a volume group
+message DeleteVolumeGroupRequest {
+  // The ID of the volume group to be deleted.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Secrets required by the driver to complete the
+  // delete request.
+  string secret_name = 2;
+  string secret_namespace = 3;
+}
+// DeleteVolumeGroupResponse holds the information to send when
+// volumeGroup is successfully deleted.
+message DeleteVolumeGroupResponse {
+}
+// ModifyVolumeGroupMembershipRequest holds the required
+// information to modify a volume group
+message ModifyVolumeGroupMembershipRequest {
+  // The ID of the volume group to be modified.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Specify volume_ids that will be in the modified volume group.
+  // This list will be compared with the volume_ids in the existing
+  // group.
+  // New ones will be added and missing ones will be removed.
+  // If no volume_ids are provided, all existing volumes will
+  // be removed from the group.
+  // This field is OPTIONAL.
+  repeated string volume_ids = 2;
+
+  // Secrets required by the driver to complete the
+  // modification request.
+  string secret_name = 3;
+  string secret_namespace = 4;
+}
+// ModifyVolumeGroupMembershipResponse holds the information to
+// send when volumeGroup is successfully modified.
+message ModifyVolumeGroupMembershipResponse {
+  // Contains the identifier of the modified volume group.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
```

Comment:
this looks pretty useless? Why would the `volume_group_id` need to be returned after the contents of the group was modified?

Reply:
Makes sense, we can remove that. Anyway, I think the volume_group_id would be immutable once created.

Reply:
The actual spec is to fetch the complete volumegroup using the group id, but since we are not using that as of now that's the reason it seems redundant for now.
We might probably update the code in future to fetch the volume group and do some assertions/use data from it, so maybe we can let this be like this for now. WDYT?

Reply:
I think you have two options, it is up to you to decide what you like to do:

1. return the complete VolumeGroup object and inspect/verify the results
2. use an empty message, and rely on an error in case something went wrong